examples/basic-infection/README.md

```diff
@@ -183,82 +186,130 @@ infection status:
+        );
     }
-    context.observe_person_property_event(update_property_counter);
-    context.add_plan(report_periodic_item, time = 0);
+    context::observe_person_property_event(update_property_counter);
+    context::on_period_end(0, report_periodic_item);
 }
 ```
-Methods are implemented for person property counter to increment and decrement the counters. For the periodic report, one method will release the periodic items to a report based on the period defined in parameters. Changes in the person properties are observed and the callback function `update_property_counter` updates the counts for each property.
-```
-fn update_property_counter(context, person_id,
-                           person_property_type,
-                           previous_person_property_value) {
-    person_property_counter.increment(context.get_person_property(person_property_type, person_id));
-    person_property_counter.decrement(person_property_type, previous_person_property_value);
+Methods are implemented for person property counter to increment and decrement
+the counters. Changes in the person properties are
+observed and the callback function `update_property_counter` updates the
+counts for each property:
+
+```rust
+fn handle_infection_status_change(context, person_id, prev, current) {
+    counter::increment(current);
+    counter::decrement(prev);
 }
+```
 
-fn report_periodic_item(context) {
-    // Report all items in the counter
-    for person_property in context.get_person_property_types().to_list() {
-            for person_property_value in (person_property.to_list()) {
-                report_item = {
-                t = context.get_time(),
-                person_property_type = person_property,
-                person_property_value = person_property_value,
-                person_property_counter = person_property_counter.get_counter(person_property, person_property_value),
-                }
-                context.print_report_data(report_item);
-            }
-        }
+When all plans have executed for a given time t, `ixa` calls the on_period_end
+callback to write the report row and schedule the next periodic report:
+
+```rust
+fn report_periodic_item(day, context) {
+    // Add a row for each status
+    for infection_status in counter::iter() {
+        context.add_report(Period(
+            day,
+            infection_status,
+            count: counter::get_count(infection_status)
+        ));
     }
 
     // Schedule next report day
-    next_report_time = context.get_time() + paramters.get_parameter(reporting_period);
-    if next_report_time < parameters.get_parameter(max_days) {
-        context.add_plan(report_periodic_item, next_report_time);
+    next_report_time = context::get_time() + parameters::get_parameter(reporting_period);
+    if next_report_time < parameters::get_parameter(max_days) {
+        context::plans_did_execute(report_periodic_item, report_periodic_item);
```

Comment:
what does `plan_did_excute` do?

Reply:
Ah that's supposed to be `on_period_end`